packages/runtime/container-runtime/src/opLifecycle/README.md

```diff
@@ -8,17 +22,38 @@ There are two features which can be used to work around this size limit, batch c
 
 By default, the runtime is configured with a max batch size of `716800` bytes, which is lower than the 1MB limit. The reason for the lower value is to account for possible overhead from the op envelope and metadata.
 
-## Table of contents
+### How batching works
 
--   [Introduction](#introduction)
-    -   [Compression](#compression)
-    -   [Grouped batching](#grouped-batching)
-        -   [Risks](#risks)
-    -   [Chunking for compression](#chunking-for-compression)
-    -   [Disabling in case of emergency](#disabling-in-case-of-emergency)
-    -   [Example configs](#example-configs)
-    -   [How it works](#how-it-works)
-    -   [How grouped batching works](#how-grouped-batching-works)
+Batching in the context of Fluid ops is a way in which the framework groups ops and sends them to the server in a single payload. Afterwards the ops will be broadcasted in the same order to all the other connected clients. Additional logic and validation ensure that batches do not interleave and they are processed in isolation. This means two things:
+
+-   A client will always start sending a new batch only after finishing a batch. Batches will never be sent interleaved or nested.
+-   When receiving a batch of ops, the client will process it in isolation without interleaving it with other batches or ops from other clients
+
+The way batches are formed is governed by the `FlushMode` setting of the `ContainerRuntimeOptions` and it is immutable for the entire lifetime of the runtime and subsequently the container.
+
+```
+export enum FlushMode {
+    /**
+     * In Immediate flush mode the runtime will immediately send all operations to the driver layer.
+     */
+    Immediate,
+
+    /**
+     * When in TurnBased flush mode the runtime will buffer operations in the current turn and send them as a single
+     * batch at the end of the turn. The flush call on the runtime can be used to force send the current batch.
+     */
+    TurnBased,
+}
+```
+
+What this means is that `FlushMode.Immediate` will send each op it its own payload to the server, while `FlushMode.TurnBased` will accumulate all ops in a single JS turn and send them together in the same payload. Technically, `FlushMode.Immediate` can be simulated with `FlushMode.TurnBased` by calling any async API immediately after producing each op. Therefore, for all intents and purposes, `FlushMode.Immediate` enables all batches sent to the server to have only one op.
+
+**By default, Fluid uses `FlushMode.TurnBased`** as:
+
+-   it is more efficient from an I/O perspective (batching ops overall decrease the number of payloads sent to the server)
+-   reduces concurrency related bugs, as it ensures that all ops are generated within the same JS turn and then applied by all clients within a single JS turn. Clients using the same pattern can safely assume ops will be applied exactly as they are observed locally.
```

Comment:
I think this reads a bit clearer. The fact that the ops are generated within the same JS turn is not exactly _thanks_ to `turn-based`, right? If I'm understanding correctly, `Immediate` guarantees that every op will happen on a separate JS turn because it explicitly splits them (and calls async APIs to send each of them, thus causing the turn to end?). But `turn-based` just respects whatever the app did in terms of which ops were generated within the JS turn, right?
```suggestion
-   reduces concurrency related bugs, as it ensures that all ops generated within the same JS turn are also applied by all other clients within a single JS turn. Clients using the same pattern can safely assume ops will be applied exactly as they are observed locally.
```